### PR TITLE
[misc]: Remove empty genes and variants when patient saved.

### PR DESCRIPTION
**Pull request by veronikaslc**
Fixed.

### CODE REVIEW

**Comment by marta-** [comment]
@itaiGershtansky @AlexLapena please review

File: components/patient-data/impl/src/main/java/org/phenotips/data/internal/EmptyGenotypeObjectsRemover.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    @Override
+    public void onEvent(Event event, Object source, Object data)
+    {
+        XWikiDocument doc = (XWikiDocument) source;
+
+        Map<String, EntityReference> refs = new LinkedHashMap<String, EntityReference>();
+        refs.put(GENE_KEY, GENE_CLASS_REFERENCE);
+        refs.put(VARIANT_KEY, VARIANT_CLASS_REFERENCE);
```
Comment (itaiGershtansky) [inline components/patient-data/impl/src/main/java/org/phenotips/data/internal/EmptyGenotypeObjectsRemover.java]: The building of the `refs` should be done in the constructor.

File: components/patient-data/impl/src/main/java/org/phenotips/data/internal/EmptyGenotypeObjectsRemover.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+ * Removes gene and variant objects from the document if key fields are empty, gene field for genes and cdna field for
+ * variants.
+ *
+ * @version $Id$
+ * @since 1.3M2
+ */
+@Component
+@Named("empty-genotype-objects-remover")
+@Singleton
+public class EmptyGenotypeObjectsRemover extends AbstractEventListener
```
Comment (itaiGershtansky) [inline components/patient-data/impl/src/main/java/org/phenotips/data/internal/EmptyGenotypeObjectsRemover.java]: If this is a functionality we want to have for other properties as well, the class can be renamed to EmptyObjectsRemover.

**Comment by AlexLapena** [comment]
+1

File: components/patient-data/impl/src/main/java/org/phenotips/data/internal/EmptyGenotypeObjectsRemover.java (hunk 2 above)
Comment (veronikaslc) [inline components/patient-data/impl/src/main/java/org/phenotips/data/internal/EmptyGenotypeObjectsRemover.java]: We will leave this improvement for later. Other comments addressed and PR rebased.

**Pull request merged**